### PR TITLE
Devise

### PR DESCRIPTION
**Pull request by Siianko**
## PT-link: 
https://www.pivotaltracker.com/story/show/160228213

## User story:
```
As a System Owner
In order to have control of the Campaigns created
I would like my Guests to register first
```

## Tasks
- Add gem `Devise`
- Generate Devise Model `User`
- Keep validation on User model basic: email + password
- Authenticate user on `Campaign#create`
- Add FactoryBot for User

## New gems
`devise`

## New Migrations
DeviseCreateUsers

## What've learned
- How to make form respond on click by adding `respond_to :js` to a `RegistrationsController`

gif-venue-sign-up

### CODE REVIEW

File: spec/factories/users.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FactoryBot.define do
+  factory :user do
+    sequence(:email) {|u| "mikael#{u}@test.com"}    
+    password 'my-password'
```
Comment (ESteinman) [inline spec/factories/users.rb]: Add password confirmation?

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
```
Comment (faraznaeem) [inline app/models/user.rb]: remove comments

File: features/step_definitions/paths_steps.rb [hunk 3]
```diff
@@ -1,3 +1,3 @@
-When("I am on the {string} page") do |page|
-    visit root_path
-end
\ No newline at end of file
+When("I am on the landing page") do 
+    visit '/'
```
Comment (faraznaeem) [inline features/step_definitions/paths_steps.rb]: didn't `root_path` work?

File: features/step_definitions/basic_steps.rb [hunk 4]
```diff
@@ -5,3 +5,11 @@
 Then("I fill in {string} with {string}") do |field, content|
     fill_in field, with: content
 end
+
+And("I wait {int} second(s)") do |seconds|
+    sleep seconds
+end
+
+And("stop") do  
+    binding.pry
```
Comment (faraznaeem) [inline features/step_definitions/basic_steps.rb]: is this needed?

File: config/initializers/devise.rb [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+
+# Use this hook to configure devise mailer, warden hooks and so forth.
+# Many of these configuration options can be set straight in your model.
+Devise.setup do |config|
+  # The secret key used by Devise. Devise uses this key to generate
+  # random tokens. Changing this key will render invalid all existing
+  # confirmation, reset password and unlock tokens in the database.
+  # Devise will use the `secret_key_base` as its `secret_key`
+  # by default. You can change it below and use your own secret key.
+  # config.secret_key = '938a4f436dacf146059e9c84568faf728bbde02f90d7d69dada6048cd1691b5a4e2e1c39ce69aab7678164a660af41cd25044f5e4a208dd6542afad4f8f1d0e8'
```
Comment (faraznaeem) [inline config/initializers/devise.rb]: Should this be visable?

File: features/step_definitions/basic_steps.rb (hunk 4 above)
Comment (ESteinman) [reply features/step_definitions/basic_steps.rb]: Yes we will need.

**Pull request merged**